Scripts/ProcessIP2CountryData.swift

```diff
@@ -142,21 +142,30 @@ class Processor {
 
         /// Structure of the data should be `network,registered_country_geoname_id`
         let countryBlockPrefix: String = "Processing country blocks: "
+        var prevId: String? = nil
         lines[1...].enumerated().forEach { index, line in
             guard keepRunning else { return }
             
             let values: [String] = line
                 .trimmingCharacters(in: .whitespacesAndNewlines)
                 .components(separatedBy: ",")
             
+            let currId = values[1]
+            
             guard
                 values.count == 2,
                 let ipNoSubnetMask: String = values[0].components(separatedBy: "/").first,
                 let ipAsInt: Int = IPv4.toInt(ipNoSubnetMask)
             else { return }
             
-            cache.countryBlocksIPInt.append(ipAsInt)
-            cache.countryBlocksGeonameId.append(values[1])
+            if prevId == currId {
+                cache.countryBlocksIPInt[cache.countryBlocksIPInt.count - 1] = ipAsInt
```

Comment:
Was there a particular reason to update the IP value to be the largest found in each block rather than just keeping the smallest in each block? (Just wondering if this could result in country names being off by 1 when ding the lookup in the app)

Reply:
Yeah, I think if there's a list of duplicate IPs and our target is "in range" the last one should cover all the bases:

https://github.com/oxen-io/session-ios/blob/2544cd6bc8deb28c9722d27eebd1c1e91fa2f9f7/Session/Utilities/IP2Country.swift#L155

e.g.

```
1.0.0.0,100
1.0.0.1,100
1.0.0.2,200
```

We want to prefer to keep `1.0.0.1` over `1.0.0.0` because if we dropped that, we might not pass the `where` check... actually looking at that, it should probably be `{ $0 >= ipAsInt }` not `{ $0 > ipAsInt }`, as if it's an exact match, we probably want to use it 😂

```
let countryBlockGeonameIdIndex: Int = cache.countryBlocksIPInt.firstIndex(where: { $0 > ipAsInt }).map({ $0 - 1 }),
```

Reply:
Ah yea, we'd also need to remove the `.map({ $0 - 1 })` in that case

The current logic was setup a little odd (I'm sure there was a reason but it was before my time) - essentially it looks through `countryBlocksIPInt` until it finds a value _larger_ than the `ipAsInt` it's looking for (ie. the next block) and then subtracts 1 from the index it finds in order to select the correct countryIPBlock

I've added a TODO in my code to update this to your suggestion here since it's more straight forward than the current approach

Reply:
oh, right, thanks, I get a bit confused by the Swift `map()` 😂